projects/laji/src/app/shared-modules/datatable/datatable/datatable.component.ts

```diff
@@ -273,4 +309,89 @@ export class DatatableComponent implements AfterViewInit, OnInit, OnDestroy {
       this.logger.info('selected row index failed', e);
     }
   }
+
+  private sortRows(sorts: DatatableSort[] = []) {
+    if (this.sortSub) {
+      this.sortSub.unsubscribe();
+    }
+
+    if (this.clientSideSorting && this._rows) {
+      this.sortLoading = true;
+      this.sortSub = this.sort(sorts, this._rows)
+      .subscribe(() => {
+          this._rows = [...this._rows];
+          this.sortLoading = false;
+          this.changeDetectorRef.markForCheck();
+        }
+      );
+    } else {
+      this.sortLoading = false;
+    }
+  }
```

Comment:
The code is a bit had to follow since it's doing the sorting in a mutable way. Looking at this:
```
      this.sortSub = this.sort(sorts, this._rows)
      .subscribe(() => {
          this._rows = [...this._rows];
```
It looks like a pipe without any return value is called, and you kind of have to guess that the pipe sorted the `this._rows` before it's reassigned to a new variable.

Doing mutable operations is a common source of confusion and therefore bugs, and since we are anyway reassigning the mutably sorted `this._rows`, I think this would be more readable if the operations were done in an immutable way.

The rest of the comments are guidelines for how to get the sorting done immutably.

Reply:
@olzraiti Good suggestion, changes related to this should be now done